jenkins/jobs/dsl/docker_pipeline_jobs.groovy

```diff
@@ -265,24 +265,44 @@ vulnerabilityScan.with{
           buildNumber('${B}')
       }
     }
-    shell('''set +x
-            |echo "THIS STEP NEEDS TO BE UPDATED ONCE ACCESS TO A PRODUCTION CLAIR DATABASE IS AVAILABLE"
-            |
-            |if [ -z ${CLAIR_DB} ]; then
-            | echo "WARNING: You have not provided the endpoints for a Clair database, moving on for now..."
-            |else
-            | # Set up Clair as a docker container
-            | echo "Clair database endpoint: ${CLAIR_DB}"
-            | mkdir /tmp/clair_config
-            | curl -L https://raw.githubusercontent.com/coreos/clair/master/config.example.yaml -o /tmp/clair_config/config.yaml
-            | # Add the URI for your postgres database
-            | sed -i'' -e "s|options: |options: ${CLAIR_DB}|g" /tmp/clair_config/config.yaml
-            | docker run -d -p 6060-6061:6060-6061 -v /tmp/clair_config:/config quay.io/coreos/clair -config=/config/config.yaml
-            | # INSERT STEPS HERE TO RUN VULNERABILITY ANALYSIS ON IMAGE USING CLAIR API
-            |fi
-            |
-            |echo LOGIN=$(echo ${DOCKER_LOGIN}) > credential.properties
-            |set -x'''.stripMargin())
+    	shell('''#!/bin/bash
+			|set +x
+			|
+			|if [ -z ${CLAIR_DB} ]; then
+			| echo "[WARNING] You have not provided the ClairDB parameter, skipping vulnerability scan..."
+			|else
+			| # Preparing Clair configuration
+			| git clone https://github.com/coreos/clair.git
+			| cd clair
+			| git checkout tags/v1.2.6
+			| sed -i -e "s|source:|source: ${CLAIR_DB}|" config.example.yaml
+			|
+			| # Provisioning Clair container
+			| sed -i -e "s|VOLUME /config|ADD config.example.yaml /config/config.yaml|" Dockerfile
+			| docker build -t clair:v1.2.6 .
+			| docker rm -vf clair
+			| docker run --name clair -d -p 6060-6061:6060-6061 -v /var/run/docker.sock:/var/run/docker.sock clair:v1.2.6 -config=/config/config.yaml
+			|
+			| # Installing docker binaries inside of a Clair container (to get hosts docker image IDs)
```

Comment:
Basically if you will build custom image and publish on dockerhub, you can builtin docker there, as we've done this with Jenkins Slave, so no need to overwhelm provisioning steps, just pull image and use it.